File: Important Problems/13. MaximumXORSubset.js
// Initialize a variable max_xor to 0.
// Iterate over all possible subsets of the array arr[].
// For each subset, compute the XOR value of all the elements in the subset.
// If the result is greater than the current max_xor, update max_xor to the new result.
// Return max_xor.

function findMaxXOR(arr) {
  let n = arr.length;
  let max_xor = 0;

  // iterate over all possible subsets of arr[]
  for (let i = 0; i < 1 << n; i++) {
    let subset_xor = 0;
    // compute the XOR value of all elements in the current subset
    for (let j = 0; j < n; j++) {
      if ((i & (1 << j)) !== 0) {
        subset_xor ^= arr[j];
      }
    }
    // update max_xor if the current subset XOR value is greater
    if (subset_xor > max_xor) {
      max_xor = subset_xor;
    }
  }

  return max_xor;
}

console.log(findMaxXOR([2, 4, 5]));

File: 02. JavaScript/01. DSA/01. Language Wise Practice/03. Java/01. Bit Manipulation/src/13. MaximumXORSubset.java
import java.util.Scanner;

class MaximumXORSubset {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.println("Welcome to the Max Subset XOR Finder! ");
        System.out.print("Please Enter the size of the array: ");
        int n = input.nextInt();

        int[] arr = new int[n];
        System.out.println("Please Enter the elements of the array: ");
        for(int i = 0; i < n; i++){
            arr[i] =    input.nextInt();
        }
        input.close();

        int maxXor = findMaxSubsetXOR(arr);
        System.out.println("The Maximum XOR value is: " + maxXor);
    }

    public static int findMaxSubsetXOR(int[] arr){
            int n = arr.length;
            int maxXor = 0;

            // Iterate over all possible subsets
            for(int i = 0; i < (1 << n); i++){
                int subsetXor = 0;

                for(int j = 0; j < n; j++){
                    if((i & (1 << j)) != 0){
                        subsetXor ^= arr[j];
                    }
                }
                if(subsetXor > maxXor){
                    maxXor = subsetXor;
                }
            }
            return maxXor;
    }

}
